packages/runtime/lifecycle/src/api.ts

```diff
@@ -28,14 +38,177 @@ export type RuntimeLifecycle = {
+      *            "mediaType": "image/png",
+      *            "files": [
+      *                {
+      *                  "name": "icon-1000",
+      *                  "mediaType": "image/webp",
+      *                  "src": "ipfs://QmUbvavFxGSSEo3ipQf7rjrELDvXHDshWkHZSpV8CVdSE5"
+      *                }
+      *              ]
+      *          })
+      *   , "open_Role_C" : mkMintOpenRoleToken()
+      *   , "open_Role_D" : mkMintOpenRoleToken(
+      *          2, // Quantities
+      *          { "name": "open_Role_D Marlowe Role Token",
+                  "description": "These are metadata for closedRoleB",
+      *            image": "ipfs://QmaQMH7ybS9KmdYQpa4FMtAhwJH5cNaacpg4fTwhfPvcwj",
+      *            "mediaType": "image/png",
+      *            "files": [
+      *                {
+      *                  "name": "icon-1000",
+      *                  "mediaType": "image/webp",
+      *                  "src": "ipfs://QmUbvavFxGSSEo3ipQf7rjrELDvXHDshWkHZSpV8CVdSE5"
+      *                }
+      *              ]
+      *          })
+      * }
+      *
+      *  ////////////// 
+      *  // #2 Use Minted Roles Tokens 
+      *  const aUseMintedRoleTokensConfiguration =
+      *      useMintedRoles(
+      *        "e68f1cea19752d1292b4be71b7f5d2b3219a15859c028f7454f66cdf",
+      *        ["role_A","role_C"]
+      *      )
+      * ```
+      *
+      * @see
+      * - {@link @marlowe.io/runtime-rest-client!contract.useMintedRoles}
+      * - {@link @marlowe.io/runtime-rest-client!contract.mintRole}
+      * - Open Roles Runtime Implementation : https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe-runtime/doc/open-roles.md
+      */
+  rolesConfiguration?: RolesConfiguration;
+
+  /**
+   * Marlowe Tags are stored as Metadata within the Transaction Metadata under the top-level Marlowe Reserved Key (`1564`).
+   * Tags allows to Query created Marlowe Contracts via {@link @marlowe.io/runtime-rest-client!index.RestClient#getContracts | Get contracts }
+   *
+   * <h4>Properties</h4>
+   *
+   * 1. They aren't limited size-wise like regular metadata fields are over Cardano.
+   * 2. Metadata can be associated under each tag
+   *
+   * @example
+   * ```ts
+   * const myTags : Tags = { "My Tag 1 That can be as long as I want": // Not limited to 64 bytes
+   *                            { a: 0
+   *                            , b : "Tag 1 content" // Limited to 64 bytes (Cardano Metadata constraint)
+   *                            },
+   *                         "MyTag2": { c: 0, d : "Tag 2 content"}};
+   * ```
+   */
   tags?: Tags;
+  /**
+   * Cardano Metadata about the contract creation.
+   * <h4>Properties</h4>
+   * <p>
+   * Metadata can be expressed as a JSON object with some restrictions:
+   *   - All top-level keys must be integers between 0 and 2^64 - 1.
+   *   - Each metadata value is tagged with its type.
+   *   - Strings must be at most 64 bytes when UTF-8 is encoded.
```

Comment:
I'm assuming it should be "Strings must be at most 64 characters long when UTF-8 is encoded."